### ISSUE TITLE
Import fails if plugin directory is not Assets/Cradle

### ISSUE THREAD

**Issue by samneub** [daterre/Cradle#25]
I don't totally understand why it isn't working, but I imported all of the Asset files from the latest commit and then attempted to bring in an .html twine file. I got a Win32Exception, posted below. I'd also be fine exporting to the twee format since that appears to have a parser in Cradle but have been unable to find any way to do that from Twine 2. If you know of a way that would be awesome. This is a really cool project, thanks!

Win32Exception: ApplicationName='/Users/Sam/gressenhall/sams-gressenhall/Assets/Cradle/Editor/ThirdParty/PhantomJS/bin/osx/phantomjs', CommandLine='"phantom.js_" "file:///Users/Sam/gressenhall/sams-gressenhall/Assets/PatrickKennedy.html" "/Users/Sam/gressenhall/sams-gressenhall/Assets/Cradle/Editor/js/StoryFormats/Harlowe/harlowe.bridge.js_"', CurrentDirectory='/Users/Sam/gressenhall/sams-gressenhall/Assets/Cradle/Editor/js'
System.Diagnostics.Process.Start_noshell (System.Diagnostics.ProcessStartInfo startInfo, System.Diagnostics.Process process)
System.Diagnostics.Process.Start_common (System.Diagnostics.ProcessStartInfo startInfo, System.Diagnostics.Process process)
System.Diagnostics.Process.Start ()
(wrapper remoting-invoke-with-check) System.Diagnostics.Process:Start ()
Cradle.Editor.Utils.PhantomJS.Run[HarloweStoryData](System.String url, System.String bridgeScriptPath, Boolean throwExOnError) (at d:/Projects/UnityTwine/git/Assets/Cradle/.src/Cradle.Editor/Editor/Utils/PhantomJS.cs:37)
Cradle.Editor.StoryFormats.Harlowe.HarloweTranscoder.Init () (at d:/Projects/UnityTwine/git/Assets/Cradle/.src/Cradle.Editor/Editor/StoryFormats/Harlowe/HarloweTranscoder.cs:113)
Cradle.Editor.StoryImporter.Transcode () (at d:/Projects/UnityTwine/git/Assets/Cradle/.src/Cradle.Editor/Editor/StoryImporter.cs:44)
Cradle.Editor.CradleAssetProcessor.OnPostprocessAllAssets (System.String[] importedAssets, System.String[] deletedAssets, System.String[] movedAssets, System.String[] movedFromAssetPaths) (at d:/Projects/UnityTwine/git/Assets/Cradle/.src/Cradle.Editor/Editor/CradleAssetProcessor.cs:82)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono-runtime-and-classlibs/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono-runtime-and-classlibs/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:232)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at /Users/builduser/buildslave/mono-runtime-and-classlibs/build/mcs/class/corlib/System.Reflection/MethodBase.cs:115)
UnityEditor.AssetPostprocessingInternal.PostprocessAllAssets (System.String[] importedAssets, System.String[] addedAssets, System.String[] deletedAssets, System.String[] movedAssets, System.String[] movedFromPathAssets) (at /Users/builduser/buildslave/unity/build/Editor/Mono/AssetPostprocessor.cs:27)
UnityEditorInternal.InternalEditorUtility:ProjectWindowDrag(HierarchyProperty, Boolean)
UnityEditor.DockArea:OnGUI()

**Comment by samneub** [comment]
I imported the files into a subfolder which wasn't kosher apparently. So Importing into the main assets folder fixed this. I did get this new error though: "Reload Assembly called from managed code directly. This will cause a crash. You should never refresh assets in synchronous mode or enter playmode synchronously from script code."

**Comment by daterre** [comment]
What was the subfolder that failed? I'm not aware of any problems with specific subfolders.

Regarding that last error, it happens occasionally but has so far proved to be harmless.

**Comment by samneub** [comment]
I just created a Su folder for importing the assets, but that caused it to
malfunction as it broke some file paths. Good to know re the other error.
Thanks. Btw I'm working on a project to make video games in Unity for a
professor to use in their classroom. You have an awesome project, thanks!

**Comment by BrodyB** [comment]
Just want to confirm that I'm seeing the same thing with the latest version (commit d053d5d).

I'm on macOS Sierra, with Unity 5.4.0p4.

I normally put all the libraries I import into my projects into /Assets/3rd Party/. If I put the Cradle folder into the 3rd Party folder, I get the same error that samneub showed. The error goes away and Twine HTML files generate CS scripts normally if the Cradle folder is moved back to the /Assets/ folder.

**Comment by daterre** [comment]
Okay, now I understand the original problem. Thanks @BrodyB for the clarification. The problem is there is a hard-coded path to the Plugin's directory I couldn't get around it. I'll look for a workaround.

**Comment by thk123** [comment]
I'm seeing the same error but don't understand the above comments on how to fix it. Steps I have taken:

1. Cloned the current master
2. Copied the whole of Assets folder in to my Unity projects Assets folder
3. Deleted UnityTestTools, Cradle/Tests and Cradle/Examples (due to #35)
4. Stuck the exported Twine file I want to import into the Assets folder

So after all this, my directory structure looks like this:

```
UnityProjectRoot\
    Assets\
      | Cradle\
      |   | Assets\
      |   |   | AssetStoreTools\
      |   |   | Cradle\
      |   |   |   | .src\
      |   |   |   | Documentation\
      |   |   |   | Editor\
      |   |   |   | TwineTextPlayer\
      |   |   |   | Cradle.dll
      |   |   | Standard Assets\
      |   | ProjectSettings\
    ExportedTwine.html
```

Is that correct?

**Comment by BrodyB** [comment]
Sorry for the late reply.

The issue I had was I was putting the Cradle folder into a subfolder under Assets (in my case: "UnityProjectRoot\Cradle\Assets\3rd Party") and Cradle currently only works if you have it in the Assets\ folder. So, your folder structure is all good.

Make sure the ExportedTwine.html file is anywhere under the Assets\ folder (it can be in a subfolder below it, just as long as it isn't placed in a folder to or a level higher than Assets\. Cradle should automatically detect it, turn it into a C# script, and Unity will be seen compiling a new script.

**Comment by thk123** [comment]
@BrodyB No problem. I wonder why it didn't work - I was at a game jam so didn't have time to investigate properly, if I get a chance I'll try this plugin again and see what I was doing wrong.

**Comment by Senshaij** [comment]
I am seeing this same error, but following the instructions above did not seem to solve the issue. Cradle always detects the .html file, but throws the Win32 error. My folder structure looks like this:

UnityProjectRoot\
    Assets\
      | Cradle\
      |     | (All of Cradle's subfolders)
      | Scenes\
      | Twine\
      |     | ExportedTwine.html

I'm using Unity 5.5.0 with Twine 2. Any suggestions?

**Comment by Senshaij** [comment]
To anyone else who was struggling with this, I did manage to solve it:

First, I pulled the Assets subfolder out of the Cradle folder and used that Assets folder as the main Assets folder in a new project.
Then, I noticed that the new project required Unity 5.5.1, so I updated Unity.

This solved the issues I was having.

Old Folder Structure (with errors):
>UnityProjectRoot
>>Assets
>>>Scenes
>>>Twine
>>>Cradle
>>>>Assets (and all other subfolders, etc.)

New Folder Structure (fixed, no errors):
>UnityProjectRoot
>>(Other Cradle project files)
>>Assets (the one from the Cradle download)
>>>(All of Cradle's Asset files and folders)
>>>Scenes
>>>Twine
